applications/dashboard/controllers/class.profilecontroller.php

```diff
@@ -541,7 +541,14 @@ public function invitations($userReference = '', $username = '', $userID = '') {
      * - 1: Unset the force cookie and use the user agent to determine the theme.
      */
     public function noMobile($type = 'desktop') {
-        if (!Gdn::request()->isAuthenticatedPostBack(true)) {
+        // Only validate CSRF for non guests.
```

Comment:
I'd expand on this, just a little, so we remember *why* we only check authenticated users (e.g. "Only validate CSRF token for authenticated users because guests do not get the transient key cookie.")